base/commands/migration/status_stages.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	serialization2 "github.com/hazelcast/hazelcast-commandline-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type StatusStages struct {
+	migrationID              string
+	ci                       *hazelcast.ClientInternal
+	migrationsInProgressList *hazelcast.List
+	statusMap                *hazelcast.Map
+	updateTopic              *hazelcast.Topic
+	topicListenerID          types.UUID
+	updateMsgChan            chan UpdateMessage
+}
+
+func NewStatusStages() *StatusStages {
+	return &StatusStages{}
+}
+
+func (st *StatusStages) Build(ctx context.Context, ec plug.ExecContext) []stage.Stage {
+	return []stage.Stage{
+		{
+			ProgressMsg: "Connecting to the migration cluster",
+			SuccessMsg:  "Connected to the migration cluster",
+			FailureMsg:  "Could not connect to the migration cluster",
+			Func:        st.connectStage(ctx, ec),
+		},
+		{
+			ProgressMsg: "Fetching migration status",
+			SuccessMsg:  "Fetched migration status",
+			FailureMsg:  "Could not fetch migration status",
+			Func:        st.fetchStage(ctx, ec),
+		},
+	}
+}
+
+type MigrationInProgress struct {
+	MigrationID string `json:"migrationId"`
+}
+
+func (st *StatusStages) connectStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(status stage.Statuser) error {
+		var err error
+		st.ci, err = ec.ClientInternal(ctx)
+		if err != nil {
+			return err
+		}
+		st.migrationsInProgressList, err = st.ci.Client().GetList(ctx, MigrationsInProgressList)
+		if err != nil {
+			return err
+		}
+		all, err := st.migrationsInProgressList.GetAll(ctx)
+		if err != nil {
+			return err
+		}
+		m := all[0].(MigrationInProgress)
+		st.migrationID = m.MigrationID
+		st.statusMap, err = st.ci.Client().GetMap(ctx, MakeStatusMapName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateTopic, err = st.ci.Client().GetTopic(ctx, MakeUpdateTopicName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateMsgChan = make(chan UpdateMessage)
+		_, err = st.updateTopic.AddMessageListener(ctx, st.topicListener)
+		return err
+	}
+}
+
+func (st *StatusStages) fetchStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(stage.Statuser) error {
+		defer st.updateTopic.RemoveListener(ctx, st.topicListenerID)
+		for {
+			select {
+			case msg := <-st.updateMsgChan:
+				ms, err := readMigrationStatus(ctx, st.statusMap)
+				if err != nil {
+					return fmt.Errorf("reading status: %w", err)
+				}
+				ec.PrintlnUnnecessary(msg.Message)
+				if slices.Contains([]Status{StatusComplete, StatusFailed, StatusCanceled}, msg.Status) {
+					ec.PrintlnUnnecessary(fmt.Sprintf("Completion Percentage: %f", ms.CompletionPercentage))
```

Comment:
I think you should print completion percentage if migration is not complete. Not when it is complete.

Reply:
Shall we print it in all cases?

Reply:
Yeah, otherwise it will be always 100% in terminal statuses. Think of it as the progress bar described in the TDD. Instead of printing, you could have a progress bar as well.

Reply:
Also, there's something wrong in this code. You should print the completion percentage in the update log. Otherwise you will print the same thing over and over again. (you read status map once) Right now, you are printing comp. percentage in status map

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/367/commits/d5591a030703633a6e871e9362197e17f2a20a79